libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -859,6 +861,7 @@ void UARTDriver::write_pending_bytes_DMA(uint32_t n)
     dmaStreamEnable(txdma);
     uint32_t timeout_us = ((1000000UL * (tx_len+2) * 10) / _baudrate) + 500;
     chVTSet(&tx_timeout, chTimeUS2I(timeout_us), handle_tx_timeout, this);
```

Comment:
Use I method here, chVTSet call includes Syslock, could cause issues. Maybe test this with ChibiOS asserts on as well...